src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -239,7 +239,7 @@ void show() {
      */
     @FXML
     private void handleExit() {
-        raise(new ExitAppRequestEvent());
+        primaryStage.fireEvent(new WindowEvent(primaryStage, WindowEvent.WINDOW_CLOSE_REQUEST));
```

Comment:
This is most probably wrong. The ExitAppRequestEvent **must** be raised to handle logging and other functions. Check MainApp to understand why.